src/tools/jsondocck/src/directive.rs

```diff
@@ -0,0 +1,232 @@
+use std::borrow::Cow;
+
+use serde_json::Value;
+
+use crate::cache::Cache;
+
+#[derive(Debug)]
+pub struct Directive {
+    pub kind: DirectiveKind,
+    pub path: String,
+    pub lineno: usize,
+}
+
+#[derive(Debug)]
+pub enum DirectiveKind {
+    /// `//@ has <path>`
+    ///
+    /// Checks the path exists.
+    HasPath,
+
+    /// `//@ has <path> <value>`
+    ///
+    /// Check one thing at the path  is equal to the value.
+    HasValue { value: String },
+
+    /// `//@ !has <path>`
+    ///
+    /// Checks the path doesn't exist.
+    HasNotPath,
```

Comment:
It sounds a bit weird to me, needed to read the docs (yeay docs!) to get what it was about. Also, is it really worth it to have two variants for each command and its opposite? Why not having a `not: bool` argument (or field directly in the enum if you prefer?) instead?

Reply:
Because that leads to really sucky error messages. The old implementation (before #133478) did this, and it was miserable. We had to special-case looking at `not` to give a more specific error (https://github.com/rust-lang/rust/pull/82571/commits/5f24798b9e228438973a8abeee1a2d5ea016f335).

Reply:
Hum... Not completely convinced but it's fine, I also like that the type system represents better the reality instead of a field. It was just more convenient (imo) to have an `inverted` field.